Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,378 @@
+                .Where(WalletFilter(_algorithm.BrokerageModel.AccountType));
+            foreach (var item in availableWallets)
+            {
+                if (item.Available > 0)
+                {
+                    if (string.Equals(item.Currency, "USD", StringComparison.OrdinalIgnoreCase))
+                    {
+                        list.Add(new Cash(item.Currency, item.Available, 1));
+                    }
+                    else if (_symbolMapper.IsKnownFiatCurrency(item.Currency))
+                    {
+                        var symbol = Symbol.Create(item.Currency + "USD", SecurityType.Forex, Market.FXCM);
+                        var rate = GetConversionRate(symbol);
+                        list.Add(new Cash(item.Currency.ToUpper(), item.Available, rate));
+                    }
+                    else
+                    {
+                        var symbol = item.Currency + "USD";
+                        var tick = GetTick(_symbolMapper.GetLeanSymbol(symbol));
+
+                        list.Add(new Cash(item.Currency.ToUpper(), item.Available, tick.Price));
+                    }
+                }
+            }
+
+            return list;
+        }
+
+        /// <summary>
+        /// Gets the history for the requested security
+        /// </summary>
+        /// <param name="request">The historical data request</param>
+        /// <returns>An enumerable of bars covering the span specified in the request</returns>
+        public override IEnumerable<BaseData> GetHistory(Data.HistoryRequest request)
+        {
+            string resolution = ConvertResolution(request.Resolution);
+            long resolutionInMS = (long)request.Resolution.ToTimeSpan().TotalMilliseconds;
+            string symbol = _symbolMapper.GetBrokerageSymbol(request.Symbol);
+            long startMTS = (long)Time.DateTimeToUnixTimeStamp(request.StartTimeUtc) * 1000;
+            long endMTS = (long)Time.DateTimeToUnixTimeStamp(request.EndTimeUtc) * 1000;
+            string endpoint = $"v2/candles/trade:{resolution}:t{symbol}/hist?limit=1000&sort=1";
+
+            while ((endMTS - startMTS) > resolutionInMS)
+            {
+                var timeframe = $"&start={startMTS}&end={endMTS}";
+
+                var restRequest = new RestRequest(endpoint + timeframe, Method.GET);
+                var response = ExecuteRestRequest(restRequest);
+
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    throw new Exception($"BitfinexBrokerage.GetHistory: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+                }
+
+                var candles = JsonConvert.DeserializeObject<object[][]>(response.Content)
+                    .Select(entries => new Messages.Candle(entries))
+                    .ToList();
+
+                RestoreMissedCandles(candles, startMTS, endMTS, resolutionInMS);
```

Comment:
Just clarify this. Not sure it's really necessary, but on small resolution bitfinex miss some with no trades within interval, so data doesn't look continuty.

Just give you example (returns 58 bars instead of 60): https://api.bitfinex.com/v2/candles/trade:1m:tBTCUSD/hist?limit=1000&sort=1&start=1533150000000&end=1533153540000

you can see that there is no candles for 
1533152220000 (1 August 2018 г., 19:37:00)
1533153000000 (1 August 2018 г., 19:50:00)

If you think that Lean handle this itself, i'm happy to remove this trick.

Reply:
`Candles with Volume = 0 are not returned from service` - These should not be returned; LEAN automatically fills forward by default and we allow them to disable that with `fillForward:false`

Reply:
Shall I remove my own implementation? or leave it as it is?

Reply:
Unfortunately we'll have to remove the call to `RestoreMissedCandles` and the method itself